package.json

```diff
@@ -36,34 +36,34 @@
   },
   "homepage": "https://github.com/alpheios-project/wordsvc#readme",
   "devDependencies": {
-    "@actions/core": "^1.2.4",
-    "@babel/core": "^7.11.4",
-    "@babel/plugin-proposal-object-rest-spread": "^7.10.4",
-    "@babel/plugin-transform-modules-commonjs": "^7.10.4",
-    "@babel/plugin-transform-runtime": "^7.10.4",
-    "@babel/preset-env": "^7.10.4",
-    "@babel/register": "^7.10.4",
-    "@babel/runtime": "^7.10.4",
-    "alpheios-core": "git+https://github.com/alpheios-project/alpheios-core.git#production",
-    "alpheios-node-build": "github:alpheios-project/node-build#semver:^3.5.1",
-    "archiver": "^4.0.1",
+    "@actions/core": "^1.2.6",
+    "@babel/core": "^7.12.3",
+    "@babel/plugin-proposal-object-rest-spread": "^7.12.1",
+    "@babel/plugin-transform-modules-commonjs": "^7.12.1",
+    "@babel/plugin-transform-runtime": "^7.12.1",
+    "@babel/preset-env": "^7.12.1",
+    "@babel/register": "^7.12.1",
+    "@babel/runtime": "^7.12.5",
+    "alpheios-core": "git+https://github.com/alpheios-project/alpheios-core.git",
+    "alpheios-node-build": "github:alpheios-project/node-build#v4",
```

Comment:
I didn't understand why do you need to use another version here.
I believe that we have `v3` merged to master now, and I use `v3` inside `alignment-editor-new`.
And would update to `master`. 
Now I would need to support a new version - `v4`?

Reply:
That's because `webpack-merge` has been updated to the latest version, 5.3.0. The `v3` can work only with `webpack-merge` 4.x.x. It requires modifications to support 5.x.x (I don't like that but that's the way it is). So `v4` is basically `v3` modified to support the latest versions of dependencies.

Please let me know if it's OK to keep the `v4` here.

Reply:
I wonder what version should I use for my `alignment-edittor`.
I could see `webpack-merge` in my dependencies but I don't need to update it for the development process.
Should I update it too - to support the same version? What would be the strategy for using `node-build` ?

Reply:
I think the rule of thumb is:
1. If you want to use the latest version of dependencies, including `webpack-merge` 5.x, you have to use `v4.0.0` (it is a tag and I would recommend to use it over branch because it guarantees a stability of the `node-build` code; I did not use it myself because I've had some issues with npm not pulling it out and I was not able to figure it out within a reasonable amount of time) or `v4` (it is a branch).
2. If you continue to use `webpack-merge` 4.x you can use `node-build` `v3.5.1` or do not specify the version at all. In the latter case it will pull the latest version from the master which would be the same `v3.5.1`. 3.5.1 is more tested and thus I think is generally more reliable (but `v4` in the other hand did not introduce any new features, it just updated setting to work with the latest build environment, so risks are not that high).

Reply:
> I could see webpack-merge in my dependencies but I don't need to update it for the development process.
Should I update it too - to support the same version? What would be the strategy for using node-build ?

I think `webpack-merge` version is not critical as the only responsibility of this package is to merge options of webpack together. I would say it's up to you to decide which version to choose. But generally, I think, it is preferable for us to have the same (latest) versions of dependencies. The latest versions, however, may create pain points sometimes so maybe it's better to test them on a single package and then, if it worked fine, update the other packages. In majority of the cases we can update dependencies without any issues and without any need to accommodate our code, but sometimes it just does now work this way 🙂. So it's a lottery ...

Reply:
Thank you for detailed explanations. I would make some experiments tomorrow - today `npm install` has alrady become too slowly :)